File: ch08/App_Code/CS/Listing8-58.cs
using System;
using System.CodeDom;
using System.Web.Compilation;
using System.Web.UI;

namespace Wrox.CS
{
[ExpressionPrefix("MyThirdCustomExpression")]
[ExpressionEditor("MyThirdCustomExpressionEditor")]
public class MyThirdCustomExpression : ExpressionBuilder
{
    public override System.CodeDom.CodeExpression
        GetCodeExpression(BoundPropertyEntry entry, object parsedData,
            ExpressionBuilderContext context)
    {
        return new CodeCastExpression("Int64",
            new CodePrimitiveExpression(parsedData));
    }

    public override object ParseExpression
        (string expression, Type propertyType, 
         ExpressionBuilderContext context)
    {
        return expression;
    }

    public override bool SupportsEvaluate
    {
        get
        {
            return true;
        }
    }

    public override object EvaluateExpression(object target,
        BoundPropertyEntry entry, object parsedData,
        ExpressionBuilderContext context)
    {
        return parsedData;
    }
}
}

File: ch08/App_Code/VB/Listing8-58.vb
Imports System
Imports System.CodeDom
Imports System.Web.Compilation
Imports System.Web.UI

Namespace Wrox.VB


    <ExpressionPrefix("MyCustomExpression")>
    <ExpressionEditor("MyCustomExpressionEditor")>
    Public Class MyThirdCustomExpression
        Inherits ExpressionBuilder

        Public Overrides Function GetCodeExpression(ByVal entry As BoundPropertyEntry,
            ByVal parsedData As Object, ByVal context As ExpressionBuilderContext) _
            As System.CodeDom.CodeExpression

            Return New CodeCastExpression("Int64",
                New CodePrimitiveExpression(parsedData))
        End Function

        Public Overrides Function ParseExpression(ByVal expression As String,
            ByVal propertyType As Type,
            ByVal context As ExpressionBuilderContext) As Object

            Return expression
        End Function

        Public Overrides ReadOnly Property SupportsEvaluate As Boolean
            Get
                Return True
            End Get
        End Property

        Public Overrides Function EvaluateExpression(ByVal target As Object,
            ByVal Entry As BoundPropertyEntry, ByVal parsedData As Object,
            ByVal context As ExpressionBuilderContext) As Object

            Return parsedData
        End Function

    End Class
End Namespace

